components/WorkoutForm.js

```diff
@@ -0,0 +1,268 @@
+      form.elements.exerciseName.focus();
+    } else {
+      setCurrentExercises([
+        ...currentExercises,
+        { exerciseId: exerciseId, sets: sets, reps: reps },
+      ]);
+      form.elements.exerciseName.value = "";
+      form.elements.sets.value = "";
+      form.elements.reps.value = "";
+      form.elements.exerciseName.focus();
+    }
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+    const name = data.workoutName;
+    if (!currentExercises.length) {
+      alert("Please add exercises to your workout!");
+    } else {
+      onAddWorkout(name, currentExercises);
+      onCreateMode();
+      setCurrentExercises([]);
+      event.target.reset();
+    }
+  }
+
+  return (
+    <WrapperFilter>
+      <WrapperForm>
+        <FormContainer>
+          <Header>
+            <h2>Create New Workout</h2>
+          </Header>
+          <form onSubmit={handleSubmit}>
+            <Label>
+              Workout Name
+              <Input
+                type="text"
+                name="workoutName"
+                placeholder="your workout name"
+                required
+                maxLength={30}
+              />
+            </Label>
+
+            <Fieldset>
+              <Legend>New Exercise</Legend>
+              <Label>
+                Exercise Name
+                <Select name="exerciseName" defaultValue="default">
+                  <option value="default" disabled>
+                    Please select an exercise
+                  </option>
+                  {exercises.map((exercise) => (
+                    <option key={exercise.id} value={exercise.id}>
+                      {exercise.name}
+                    </option>
+                  ))}
+                </Select>
+              </Label>
+              <InlineContainer>
+                <RepsSetsLabel>
+                  Sets
+                  <Input
+                    type="number"
+                    name="sets"
+                    min="1"
+                    max="50"
+                    placeholder="min. 1"
+                  />
+                </RepsSetsLabel>
+                <RepsSetsLabel>
+                  Reps
+                  <Input
+                    type="number"
+                    name="reps"
+                    min="1"
+                    max="100"
+                    placeholder="min. 1"
+                  />
+                </RepsSetsLabel>
+              </InlineContainer>
+              <Button type="button" onClick={handleAddExercise}>
+                + Add Exercise
+              </Button>
+            </Fieldset>
+
+            {/* Display the list of added exercises */}
+            <ExerciseListDisplay>
+              <h3>Exercises</h3>
+              <ExerciseListContainer>
+                {currentExercises.map((exercise, index) => (
+                  <ExerciseItem key={index}>
```

Comment:
every exercise item seems to have an id so I would recommend using that instead of the index for the key.